processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -398,6 +430,42 @@ private void initPredicate()
     return new BoundLongPredicateSupplier();
   }
 
+  private BigDecimal getBigDecimalLowerBoundFromFloatString(String floatStr)
+  {
+    BigDecimal convertedBD;
+    try {
+      convertedBD = new BigDecimal(floatStr);
+    } catch (NumberFormatException nfe) {
+      return null;
+    }
+
+    if (lowerStrict) {
+      convertedBD = convertedBD.setScale(0, RoundingMode.FLOOR);
```

Comment:
Need not complicate control flow, suggested `return convertedBD.setScale(0, RoundingMode.FLOOR);` (and below)